### DISCUSSION TITLE
Error to install

### DISCUSSION THREAD

**Discussion by henrynicolas** [abrignoni/iLEAPP#321] [General]
HI friend 

Try to use iLEAPP in my Win11 x64 whit Phyton 3.10 x64 and run requeriments and have this error 

Microsoft Windows [Versión 10.0.22000.795]
(c) Microsoft Corporation. Todos los derechos reservados.

C:\Users\henry\OneDrive\Documentos\GitHub\iLEAPP>pip install -r requirements.txt
Ignoring python-magic: markers 'platform_system == "Linux"' don't match your environment
Ignoring python-magic-bin: markers 'platform_system == "Darwin"' don't match your environment
Collecting numpy==1.21.4
  Using cached numpy-1.21.4-cp310-cp310-win_amd64.whl (14.0 MB)
Collecting astc_decomp
  Using cached astc_decomp-1.0.3.tar.gz (58 kB)
  Preparing metadata (setup.py) ... done
Collecting biplist
  Using cached biplist-1.0.3-py3-none-any.whl
Collecting bs4
  Using cached bs4-0.0.1-py3-none-any.whl
Collecting pyliblzfse
  Using cached pyliblzfse-0.4.1.tar.gz (47 kB)
  Preparing metadata (setup.py) ... done
Requirement already satisfied: packaging==20.1 in c:\users\henry\appdata\local\programs\python\python310\lib\site-packages (from -r requirements.txt (line 6)) (20.1)
Collecting pathlib2==2.3.5
  Using cached pathlib2-2.3.5-py2.py3-none-any.whl (18 kB)
Collecting PGPy
  Using cached PGPy-0.5.4-py2.py3-none-any.whl (84 kB)
Requirement already satisfied: pillow in c:\users\henry\appdata\local\programs\python\python310\lib\site-packages (from -r requirements.txt (line 9)) (9.2.0)
Collecting pyinstaller
  Using cached pyinstaller-5.2-py3-none-win_amd64.whl (1.2 MB)
Requirement already satisfied: PySimpleGUI==4.16.0 in c:\users\henry\appdata\local\programs\python\python310\lib\site-packages (from -r requirements.txt (line 11)) (4.16.0)
Requirement already satisfied: blackboxprotobuf in c:\users\henry\appdata\local\programs\python\python310\lib\site-packages (from -r requirements.txt (line 12)) (1.0.1)
Collecting nska-deserialize>=1.3.1
  Using cached nska_deserialize-1.3.2-py3-none-any.whl
Requirement already satisfied: simplekml in c:\users\henry\appdata\local\programs\python\python310\lib\site-packages (from -r requirements.txt (line 15)) (1.3.6)
Collecting pandas
  Using cached pandas-1.4.3-cp310-cp310-win_amd64.whl (10.5 MB)
Requirement already satisfied: pycryptodome in c:\users\henry\appdata\local\programs\python\python310\lib\site-packages (from -r requirements.txt (line 17)) (3.15.0)
Requirement already satisfied: python-magic-bin==0.4.14 in c:\users\henry\appdata\local\programs\python\python310\lib\site-packages (from -r requirements.txt (line 19)) (0.4.14)
Requirement already satisfied: pyparsing>=2.0.2 in c:\users\henry\appdata\local\programs\python\python310\lib\site-packages (from packaging==20.1->-r requirements.txt (line 6)) (3.0.9)
Requirement already satisfied: six in c:\users\henry\appdata\local\programs\python\python310\lib\site-packages (from packaging==20.1->-r requirements.txt (line 6)) (1.16.0)
Requirement already satisfied: beautifulsoup4 in c:\users\henry\appdata\local\programs\python\python310\lib\site-packages (from bs4->-r requirements.txt (line 4)) (4.8.2)
Collecting pyasn1
  Using cached pyasn1-0.4.8-py2.py3-none-any.whl (77 kB)
Collecting cryptography>=2.6
  Using cached cryptography-37.0.4-cp36-abi3-win_amd64.whl (2.4 MB)
Requirement already satisfied: setuptools in c:\users\henry\appdata\local\programs\python\python310\lib\site-packages (from pyinstaller->-r requirements.txt (line 10)) (58.1.0)
Collecting pefile>=2022.5.30
  Using cached pefile-2022.5.30-py3-none-any.whl
Collecting pyinstaller-hooks-contrib>=2021.4
  Using cached pyinstaller_hooks_contrib-2022.8-py2.py3-none-any.whl (239 kB)
Requirement already satisfied: pywin32-ctypes>=0.2.0 in c:\users\henry\appdata\local\programs\python\python310\lib\site-packages (from pyinstaller->-r requirements.txt (line 10)) (0.2.0)
Collecting altgraph
  Using cached altgraph-0.17.2-py2.py3-none-any.whl (21 kB)
Requirement already satisfied: protobuf==3.10.0 in c:\users\henry\appdata\local\programs\python\python310\lib\site-packages (from blackboxprotobuf->-r requirements.txt (line 12)) (3.10.0)
Collecting python-dateutil>=2.8.1
  Using cached python_dateutil-2.8.2-py2.py3-none-any.whl (247 kB)
Requirement already satisfied: pytz>=2020.1 in c:\users\henry\appdata\local\programs\python\python310\lib\site-packages (from pandas->-r requirements.txt (line 16)) (2022.1)
Requirement already satisfied: cffi>=1.12 in c:\users\henry\appdata\local\programs\python\python310\lib\site-packages (from cryptography>=2.6->PGPy->-r requirements.txt (line 8)) (1.15.1)
Collecting future
  Using cached future-0.18.2-py3-none-any.whl
Requirement already satisfied: soupsieve>=1.2 in c:\users\henry\appdata\local\programs\python\python310\lib\site-packages (from beautifulsoup4->bs4->-r requirements.txt (line 4)) (2.3.2.post1)
Requirement already satisfied: pycparser in c:\users\henry\appdata\local\programs\python\python310\lib\site-packages (from cffi>=1.12->cryptography>=2.6->PGPy->-r requirements.txt (line 8)) (2.21)
Building wheels for collected packages: astc_decomp, pyliblzfse
  Building wheel for astc_decomp (setup.py) ... error
  error: subprocess-exited-with-error

  × python setup.py bdist_wheel did not run successfully.
  │ exit code: 1
  ╰─> [9 lines of output]
      C:\Users\henry\AppData\Local\Programs\Python\Python310\lib\site-packages\setuptools\dist.py:717: UserWarning: Usage of dash-separated 'description-file' will not be supported in future versions. Please use the underscore name 'description_file' instead
        warnings.warn(
      C:\Users\henry\AppData\Local\Programs\Python\Python310\lib\site-packages\setuptools\dist.py:717: UserWarning: Usage of dash-separated 'long-description' will not be supported in future versions. Please use the underscore name 'long_description' instead
        warnings.warn(
      running bdist_wheel
      running build
      running build_ext
      building 'astc_decomp' extension
      error: Microsoft Visual C++ 14.0 or greater is required. Get it with "Microsoft C++ Build Tools": https://visualstudio.microsoft.com/visual-cpp-build-tools/
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for astc_decomp
  Running setup.py clean for astc_decomp
  Building wheel for pyliblzfse (setup.py) ... error
  error: subprocess-exited-with-error

  × python setup.py bdist_wheel did not run successfully.
  │ exit code: 1
  ╰─> [5 lines of output]
      running bdist_wheel
      running build
      running build_ext
      building 'liblzfse' extension
      error: Microsoft Visual C++ 14.0 or greater is required. Get it with "Microsoft C++ Build Tools": https://visualstudio.microsoft.com/visual-cpp-build-tools/
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for pyliblzfse
  Running setup.py clean for pyliblzfse
Failed to build astc_decomp pyliblzfse
Installing collected packages: pyliblzfse, pyasn1, biplist, astc_decomp, altgraph, python-dateutil, pyinstaller-hooks-contrib, pathlib2, numpy, nska-deserialize, future, pefile, pandas, cryptography, bs4, pyinstaller, PGPy
  Running setup.py install for pyliblzfse ... error
  error: subprocess-exited-with-error

  × Running setup.py install for pyliblzfse did not run successfully.
  │ exit code: 1
  ╰─> [5 lines of output]
      running install
      running build
      running build_ext
      building 'liblzfse' extension
      error: Microsoft Visual C++ 14.0 or greater is required. Get it with "Microsoft C++ Build Tools": https://visualstudio.microsoft.com/visual-cpp-build-tools/
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: legacy-install-failure

× Encountered error while trying to install package.
╰─> pyliblzfse

note: This is an issue with the package mentioned above, not pip.
hint: See above for output from the failure.

**Comment by abrignoni** [comment]
Use python 3.9 per the readme.
Waiting for a few third party libraries to be updated to python 3.10.

<img width="248" alt="Screen Shot 2022-07-26 at 12 11 35 PM" src="https://user-images.githubusercontent.com/28718987/181057113-f391570e-e744-4f51-8161-43ede4d8cacf.png">

**Reply by henrynicolas** [reply]
thanks try again whit older python
